shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCallingReport.scala

```diff
@@ -0,0 +1,110 @@
+package nl.lumc.sasc.biopet.pipelines.shiva
+
+import java.io.{ File, PrintWriter }
+
+import nl.lumc.sasc.biopet.utils.rscript.LinePlot
+import nl.lumc.sasc.biopet.utils.summary.db.SummaryDb
+import nl.lumc.sasc.biopet.utils.summary.db.SummaryDb.{ ModuleName, PipelineName, SampleName }
+
+import scala.concurrent.Await
+import scala.concurrent.duration.Duration
+
+object ShivaSvCallingReport {
+
+  val histogramBinBoundaries: Array[Int] = Array(100, 1000, 10000, 100000, 1000000, 10000000)
+  val histogramPlotTicks: Array[Int] = Array(100, 1000, 10000, 100000, 1000000, 10000000, 100000000)
+  val histogramText: List[String] = List("<=100bp", "0.1-1kb", "1-10kb", "10-100kb", "0.1-1Mb", "1-10Mb", ">10Mb")
+
+  def parseSummaryForSvCounts(summary: SummaryDb, runId: Int, sampleNames: Seq[String]): Map[String, Map[String, Array[Long]]] = {
+    var delCounts, insCounts, dupCounts, invCounts: Map[String, Array[Long]] = Map()
+
+    for (sampleName <- sampleNames) {
+      val sampleCounts: Map[String, Any] = Await.result(summary.getStat(runId, PipelineName("shivasvcalling"), ModuleName("parse_sv_vcf"), SampleName(sampleName)), Duration.Inf).get
+      for ((svType, counts) <- sampleCounts.collect({ case (k, v: List[_]) => (k, v.toArray[Any]) })) {
+        val elem: Tuple2[String, Array[Long]] = (sampleName, counts.collect({ case x: Long => x }))
+        svType match {
+          case "DEL" => delCounts += elem
+          case "INS" => insCounts += elem
+          case "DUP" => dupCounts += elem
+          case "INV" => invCounts += elem
+        }
+      }
+    }
+
+    var result: Map[String, Map[String, Array[Long]]] = Map()
+    if (delCounts.exists(elem => (elem._2.sum > 0))) result = Map("DEL" -> delCounts)
```

Comment:
I figured that it's better not to show anything for a particular sv type when Biopet hasn't been configured to detect it. Take for example inversions. With the default configuration these will never be reported, by default all tools will be run (except for pindel), Pysvtools is used for merging and as one of the tools reports only dels and ins, then no inversions in the output. I considered it incorrect to report then, that in the sample being analyzed there were no inversions found.

Which sv types are shown could be decided based on Shiva's conf, but for that I'd need to know more details about what this Pysvtools is doing. I remember from testing that it can for example combine variants that have been marked with different svtype tag in the vcf-file, for example combining insertions and duplications. So I skipped studying Pysvtools and used a simpler approach, assuming that when one particular sv type wasn't detected in any of the samples, then one of the tools didn't report that type and then it can't be in the output.

You think the zeros should be present there all the time? Or it's worth writing code to make this decision based on configuration?